src/TMap.cpp

```diff
@@ -1998,61 +2097,27 @@ int TMap::createMapImageLabel(int area, QString imagePath, float x, float y, flo
     lp.drawPixmap(QPoint(0, 0), imagePixmap.scaled(drawRect.size().toSize()));
     label.size = QSizeF(width, height);
     label.pix = pix;
-    if (!mpRoomDB->getArea(area)) {
-        return -1;
-    }
 
-    int label_id;
+    int label_id = -1;
+    do {
+    } while (pA->mMapLabels.contains(++label_id));
 
-    // No labels exist for this area, so start from zero.
-    if (!mapLabels.contains(area)) {
-        QMap<int, TMapLabel> m;
-        label_id = 0;
-        m[label_id] = label;
-        mapLabels[area] = m;
-    } else {
-        label_id = createMapLabelID(area);
-        if (label_id > -1) {
-            mapLabels[area].insert(label_id, label);
-        }
-    }
+    pA->mMapLabels.insert(label_id, label);
```

Comment:
Same with this

Reply:
Well the outer `if` is no longer relevant and a negative id is not actually invalid as a key for a `QMap<int, ...>` - and if we get to the stage that we do have more than **2147483647** map labels I expect that Mudlet will already have crashed and burned...!

Reply:
No harm in being safer about it instead of just assuming Mudlet should crash at some point, that is not how we build this...